### PR TITLE
Implement inline images

### PR DESCRIPTION
**Pull request by marcusklaas**
Raises the number of passing tests to 488/633. I think there's only 1 relevant test that currently isn't passing as it's particularly nasty:

```md
![[[foo](uri1)](uri2)](uri3)
```
which should produce
```html
<p><img src="uri3" alt="[foo](uri2)" /></p>
```

I haven't figured out a way to solve in a way that's not *too* appalling. Grateful for any pointers!

### CODE REVIEW

**Comment by raphlinus** [comment]
Yeah, this nested stuff is going to be a bit tricky. I'm not sure I have much in the way of pointers, except to note that the logic for completing a link clears the link stack (`link_stack.clear()`) but the logic for an image shouldn't, because a link body can contain an image. I haven't dug into your code yet to see whether that's the problem.

**Comment by marcusklaas** [comment]
Yeah that's exactly what I'm doing :sweat_smile:

**Comment by marcusklaas** [comment]
Sounds good to me :-)

File: src/prototype.rs [hunk 1]
```diff
@@ -766,11 +768,16 @@ fn parse_line(tree: &mut Tree<Item>, s: &str, mut ix: usize) -> (usize, Option<I
                 begin_text = ix;
             }
             b'[' => {
-                tree.append_text(begin_text, ix);
+                let (link_type, start) = if ix >= begin_text + 1 && bytes[ix - 1] == b'!' {
```
Comment (ScottAbbey) [inline src/prototype.rs]: This is currently causing a panic on spec_test_573:

```
\![foo]

[foo]: /url "title"
```

[Published as #564](https://spec.commonmark.org/0.28/#example-564)

Looking backwards by bytes in this "safe" ASCII-in-UTF-8 environment isn't quite as straightforward as looking forward by bytes.  In this case, if there are an odd number of `\` before the `!`, it's a link, but if there are an even number of `\` before, it's an image.

Comment (marcusklaas) [reply src/prototype.rs]: Thanks for the heads-up! I hadn't considered that. I will try to issue a fix :-)

**Pull request merged**